components/internal/ApplicantStatusBadge.tsx

```diff
@@ -0,0 +1,133 @@
+import { Badge, Box } from '@chakra-ui/react'; // Chakra UI
+import Image from 'next/image'; // Optimized images
+
+type Props = {
+  variant:
+    | 'completed'
+    | 'inProgress'
+    | 'pending'
+    | 'rejected'
+    | 'expiring'
+    | 'expired'
+    | 'active'
+    | 'inactive';
```

Comment:
I think we're missing the `inactive` variant in the `switch` below?

Reply:
The inactive badge doesn't have an icon and so you would use a normal badge to create it
```tsx
<Badge variant="inactive">Inactive</Badge>
```
But I should remove `inactive` from the union of strings (and this explains why a final else/default was required, if I remove it I no longer need to specify that).

Reply:
Hm, we have all of the other variants encompassed in this `ApplicantStatusBadge` component, so I feel like we should include the `inactive` variant as well, even though it is trivial to render it using the `Badge` component. This gives us consistency and a single source of truth for the rendering of applicant status badges.

Reply:
Yep that makes sense, the only confusing part that led me to leave it out is that in the designs we can have an active badge with an icon and one without an icon (image below) and I wasn't sure if I should make 2 variants for that. So my thinking was anything with an icon would be in `ApplicantStatusBadge` otherwise it would use a normal badge. I could make another variant for an active badge without an icon and then have all the badges in `ApplicantStatusBadge` there if that makes it more consistent. Thoughts?

![image](https://user-images.githubusercontent.com/51224641/125226139-82142300-e29e-11eb-86c3-6301124b8bc8.png)

Reply:
That's a really good point. In that case, let's just go with your idea - leave the badges without icons to be implemented using the Chakra UI `Badge` component.

Furthermore, the badges with icons are used for requests whereas those without icons are used for applicants. Could you please rename this component to `RequestStatusBadge` instead? Apologies for this being so late.

Reply:
All good! Addressed!